### PR TITLE
Fix #142 - implement sysinfo

### LINKED ISSUE #142: Add `sysinfo` verb
Right now I have a standalone `sysinfo` program (https://github.com/NAICNO/Jobanalyzer/blob/main/code/sysinfo/sysinfo.go) that queries the configuration of the node and prints the data on stdout (as json), from where it is forwarded to the analysis machine.  On the ML nodes it is now run on boot + once a day.  I would like to have fewer programs to upgrade on all the nodes (currently there are three + a script and I'd like to reduce that to one + a script) and I think folding `sysinfo` as a verb into sonar makes a lot of sense.  It would not be much code (source or binary), we have most of the bits already.

### PR DESCRIPTION
**Pull request by lars-t-hansen**
This seems to be feature complete and working, I just need to make a final pass before asking for review.

### CODE REVIEW

**Comment by lars-t-hansen** [comment]
Also some test cases would be good.

File: src/nvidia.rs [hunk 1]
```diff
@@ -7,31 +7,76 @@
...
-    pub mem_size_kib: usize,   // Memory use in KiB /for this sample/, _not_ zero for zombies
-    pub command: String,       // The command, _unknown_ for zombies, _noinfo_ if not known
-}
+// `nvidia-smi -a` dumps a lot of information about all the cards in a semi-structured form,
+// each line a textual keyword/value pair.
+//
+// "Product Name" names the card.  Following the string "FB Memory Usage", "Total" has the
+// memory of the card.
+//
+// Parsing all the output lines in order yield the information about all the cards.
```
Comment (lars-t-hansen) [inline src/nvidia.rs:24]: "yields"

File: src/sysinfo.rs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+use serde::Serialize;
+
+use std::io::{self, Write};
+
+pub fn show_system(timestamp: &str) {
+    let fs = procfsapi::RealFS::new();
+    match do_show_system(&fs, timestamp) {
+        Ok(_) => {}
+        Err(e) => {
+            println!("FAILED: {e}");
```
Comment (lars-t-hansen) [inline src/sysinfo.rs:16]: Should use normal log channel, not stdout!!

File: src/sysinfo.rs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        // Merge equal cards
+        let mut i = 0;
+        let mut gpu_desc = "".to_string();
+        while i < cards.len() {
+            let first = i;
+            while i < cards.len() && cards[i] == cards[first] {
+                i += 1;
+            }
+            let memsize = if cards[first].mem_size_kib > 0 {
+                format!("{}", cards[first].mem_size_kib * 1024 / GIB as i64)
```
Comment (lars-t-hansen) [inline src/sysinfo.rs:74]: .to_string() instead of format?

**Comment by bast** [comment]
Thank you!

**Pull request merged**